java/src/main/java/ai/rapids/cudf/RmmEventHandler.java

```diff
@@ -22,9 +22,28 @@ public interface RmmEventHandler {
   /**
    * Invoked on a memory allocation failure.
    * @param sizeRequested number of bytes that failed to allocate
+   * @deprecated deprecated in favor of onAllocFailure(long, boolean)
    * @return true if the memory allocation should be retried or false if it should fail
    */
-  boolean onAllocFailure(long sizeRequested);
+  default boolean onAllocFailure(long sizeRequested) {
+    // this should not be called since it was the previous interface,
+    // and it was abstract before.
+    return false;
```

Comment:
Since it should not be called, we may throw an exception instead, so we can know that there was something wrongly called and reached here.

Reply:
Sure, I can make this throw. Note that this is going away very soon, it is mainly to give us a few days while we update the rest of the code. The end result is that we'll remove these default implementations, and callers must use the non-deprecated code.